org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsd/contentmodel/CMXSDElementDeclaration.java

```diff
@@ -309,6 +312,32 @@ private void collectElementsDeclaration(XSComplexTypeDefinition typeDefinition,
 		}
 	}
 
+	public boolean isOptional(String childElementName) {
+		if (!elementOptionality.containsKey(childElementName)) {

```

Comment:
I think it is better to initialize elementOptionality  to null in the constructor, because you will initialize elementOptionality only if it is null. As elementOptionality   could be empty after initilization, in your case you will do the initialization every time.